File: frontend/src/types/organization.ts
import { StandardResponse } from "./dashboard";

export interface OrganizationGoal {
  id: string;
  title: string;
  description: string;
  status: string;
  target_date: string;
}

export interface OrganizationRole {
  id: string;
  title: string;
  department: string;
  headcount: number;
}

export interface OrganizationSkill {
  id: string;
  name: string;
  category: string;
  demand_score: number;
}

export interface OrganizationExecution {
  execution_id: string;
  agent_id: string;
  status: string;
  started_at: string;
  completed_at: string | null;
}

export interface OrganizationPolicy {
  id: string;
  name: string;
  description: string;
  enforced: boolean;
}

export interface OrganizationLearning {
  id: string;
  topic: string;
  insight: string;
  confidence: number;
  discovered_at: string;
}

export interface OrganizationMetric {
  metric_name: string;
  value: number;
  trend: 'up' | 'down' | 'stable';
}

export interface OrganizationActivity {
  event_id: string;
  timestamp: string;
  actor: string;
  action: string;
  target: string;
}

File: backend/src/application/schemas/organization.py
from pydantic import BaseModel
from typing import List
from datetime import datetime

class OrganizationGoalDTO(BaseModel):
    id: str
    title: str
    description: str
    status: str
    target_date: datetime

class OrganizationRoleDTO(BaseModel):
    id: str
    title: str
    department: str
    headcount: int

class OrganizationSkillDTO(BaseModel):
    id: str
    name: str
    category: str
    demand_score: float

class OrganizationExecutionDTO(BaseModel):
    execution_id: str
    agent_id: str
    status: str
    started_at: datetime
    completed_at: datetime | None

class OrganizationPolicyDTO(BaseModel):
    id: str
    name: str
    description: str
    enforced: bool

class OrganizationLearningDTO(BaseModel):
    id: str
    topic: str
    insight: str
    confidence: float
    discovered_at: datetime

class OrganizationMetricDTO(BaseModel):
    metric_name: str
    value: float
    trend: str # 'up', 'down', 'stable'

class OrganizationActivityDTO(BaseModel):
    event_id: str
    timestamp: datetime
    actor: str
    action: str
    target: str
